includes/Data/Patterns.php

```diff
@@ -0,0 +1,76 @@
+<?php
+namespace NewfoldLabs\WP\Module\Onboarding\Data;
+
+final class Patterns {
+
+	protected static $theme_step_patterns = array(
+		'yith-wonder' => array(
+			'theme-styles' => array(
+				'site-header-left-logo-navigation-inline',
+				'homepage-1',
+				'site-footer',
+			),
+		),
+	);
+
+	public static function cleanup_wp_grammar( $content ) {
+
+		// Remove template-part if that exists
+		$content = preg_replace( '/^<!-- wp:template-part .* \/-->$/m', '', $content );
+
+		// Create an array with the values you want to replace
+		$searches = array( "\n", "\t" );
+
+		// Replace the line breaks and tabs with a empty string
+		$content = str_replace( $searches, '', $content );
+
+		return $content;
+	}
+
+	public static function get_pattern_from_slug( $pattern_slug ) {
+		$active_theme = ( \wp_get_theme() )->get( 'TextDomain' );
+		$pattern_name = $active_theme . '/' . $pattern_slug;
+
+        $block_patterns_registry = \WP_Block_Patterns_Registry::get_instance();
+		if ( $block_patterns_registry->is_registered( $pattern_name ) ) {
+			$pattern = $block_patterns_registry->get_registered( $pattern_name );
+			return array(
+				'title'   => $pattern['title'],
+				'content' => self::cleanup_wp_grammar( $pattern['content'] ),
+				'name'    => $pattern['name'],
+			);
+		}
+
+		return false;
+	}
+
+	public static function get_theme_step_patterns_from_step( $step, $squash = false ) {
```

Comment:
Perhaps a more generic name? 🙃 like the one used in **get_pattern_from_slug** Just a suggestion.

Reply:
The function name describes exactly what it does, it returns `theme_step_patterns` which is our internal map of `themes->step->patterns` given a `step` identifier. It doesn't take a `slug` and return a `pattern`.